### PR TITLE
Introduced trial-based event splitting and n-dim-mindist estimation

### PR DESCRIPTION
**Pull request by spolakh**
Work in progress. Refactoring and mining stats.

### CODE REVIEW

File: debug/fetch_data.c [hunk 1]
```diff
@@ -28,53 +28,69 @@ static const char* getfield(char* line, int num)
...
+    // Eat-up both csv headers
+    if (fgets(buf, BUF_SIZE, dataf) == NULL) FEXIT(series, series_size);
+    if (fgets(buf, BUF_SIZE, eventsf) == NULL) FEXIT(series, series_size);
+    while (fgets(buf, BUF_SIZE, eventsf)) {
+        char *tmp = strdup(buf);
+        if (strcmp(getfield(tmp, evid+2), "0") == 0) {
+            // No activity on event of interest registered -> skip
+            new = 1;
             free(tmp);
+            continue;
```
Comment (spolakh) [inline debug/fetch_data.c]: HUGE BUG up here -- I don't eat up corresponding line in dataf by continue'ing here. All results invalidated.

**Comment by spolakh** [comment]
Fixed plotting, actual charts available here: http://people.mozilla.org/~vsterzhanov/plot.html (they bring somewhat more hope to our ranks).
Added mindist cross-check, current results: Maximum matching mindist == 34.126566 VS                 Minimum nonmatching mindist == 12.842546. (w == 10, c == 3).
Going for 
1) More thorough analysis of results (percentiles rather than simple min-max)
2) SAX indexing
3) Person-wise grid searching for optimal w and c based on 1) powered by 2)

File: debug/fetch_data.c [hunk 2]
```diff
@@ -27,57 +27,164 @@ static const char* getfield(char* line, int num)
...
+    size_t n_series;
+};
+
+struct session {
+    double **series;
+    size_t series_count;
+};
+
+#define PUSH_BACK(series, size, value) \
+    (series) = realloc((series), (++(size)) * sizeof(*series)); \
```
Comment (trink) [inline debug/fetch_data.c]: - don't use a macro
- check the return value of realloc
- only assign on success

**Comment by trink** [comment]
Lets talk about this at our 1x1

**Pull request merged**